src/samples/nodejs/uart_sample.js

```diff
@@ -0,0 +1,130 @@
+/*
+ * This file is part of the Soletta Project
+ *
+ * Copyright (C) 2015 Intel Corporation. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/*
+ * This sample code demonstrates the usage of UART JS bindings for
+ * reading and writing data to a UART accessible device.
+ *
+ * Setup:
+ * Use two UART cables and connect RX/TX pins in the cable #1 to
+ * TX/RX pins in the cable #2.
+ */
+
+var uart = require( "soletta/uart" ),
+    args = process.argv.slice( 2 ),
+    uartproducer = null,
+    uartConsumer = null;
+
+var options = {
+    help: false,
+    producer,
+    consumer
+};
+
+const usage = "Usage: node uart_sample.js -p <producerUART> -c <consumerUART>\n" +
+    "options: \n" +
+    "  -h, --help \n" +
+    "  -p, --producer <producerUART>\n" +
+    "  -c, --consumer <consumerUART>\n";
+
+for ( var i = 0; i < args.length; i++ ) {
+    var arg = args[ i ];
+
+    switch ( arg ) {
```

Comment:
There are several modules available for parsing the options. I'm not using them here just to avoid the dependency.